src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/CreateGithubIssueTest.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Octokit;
+using Xunit;
+
+namespace DependencyUpdateErrorProcessor.Tests
+{
+    public class CreateGithubIssueTest : DependencyUpdateErrorProcessorTests
+    {
+        [Theory]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "ProcessPendingUpdatesAsync", "no arguments" , "2200/1/1" , false)]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "UpdateAssetsAsync", "[\"ee8cdcfb-ee51-4bf3-55d3-08d79538f94d\",\"UpdateAssetsAsync\"]",  "2200/1/1", false)]
+        public async Task ShouldCreateIssue(string repoUrl, string branch , string method, string arguments , DateTime errorOccurredAt, bool success)
+        {
+            Mock<RepositoryBranchUpdateHistoryEntry> repositoryBranchUpdate =
+                new Mock<RepositoryBranchUpdateHistoryEntry>();
+            repositoryBranchUpdate.Object.Repository = repoUrl;
+            repositoryBranchUpdate.Object.Branch = branch;
+            repositoryBranchUpdate.Object.Method = method;
+            repositoryBranchUpdate.Object.Timestamp = errorOccurredAt;
+            repositoryBranchUpdate.Object.Arguments = arguments;
+            repositoryBranchUpdate.Object.Success = success;
+            Mock<Octokit.Repository> repository = new Mock<Repository>();
+            GithubClient.Setup(x => x.Repository.Get(It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(repository.Object);
+            Mock<Maestro.Data.Models.Subscription> subscription = new Mock<Maestro.Data.Models.Subscription>();
+            subscription.Object.Id = Guid.Parse("ee8cdcfb-ee51-4bf3-55d3-08d79538f94d");
+            Context.Subscriptions.Add(subscription.Object);
+            Context.SaveChanges();
+            Mock<Octokit.Issue> issue = new Mock<Issue>();
+            GithubClient.Setup(x => x.Issue.Create(It.IsAny<long>(),It.IsAny<NewIssue>())).ReturnsAsync(issue.Object);
+            Context.RepoBranchUpdateInMemory = new List<RepositoryBranchUpdateHistoryEntry>
+                {repositoryBranchUpdate.Object};
+            DependencyUpdateErrorProcessor errorProcessor =
+                ActivatorUtilities.CreateInstance<DependencyUpdateErrorProcessor>(Scope.ServiceProvider,
+                    Context);
+            await errorProcessor.ProcessDependencyUpdateErrorsAsync();
+            GithubClient.Verify(x => x.Issue.Create(It.IsAny<long>(), It.IsAny<NewIssue>()), Times.Once);
```

Comment:
Do we care what the values of the long and the NewIssue are? It seems like we might want to verify at least a few things bou them.

Reply:
Long here is the repoId, we don't really care about that. But for NewIssue, I'm testing the contents of that in the third test -> CreateIssue